pkgs/applications/networking/cluster/waypoint/default.nix

```diff
@@ -1,46 +1,59 @@
-{ lib, buildGoModule, fetchFromGitHub, go-bindata }:
-
+{ buildGoPackage, buildGoModule, fetchFromGitHub, lib }:
+
+let
+  # nixpkgs go-bindata causes issues with the waypoint-entrypoint
+  go-bindata = buildGoPackage rec {
+    pname = "go-bindata";
+    version = "3.22.0";
+    src = fetchFromGitHub {
+      owner = "kevinburke";
+      repo = pname;
+      rev = "v${version}";
+      sha256 = "10dq77dml5jvvq2jkdq81a9yjg7rncq8iw8r84cc3dz6l9hxzj0x";
+    };
+
+    goPackagePath = "github.com/kevinburke/go-bindata";
+    subPackages = [ "go-bindata" ];
+  };
+in
 buildGoModule rec {
   pname = "waypoint";
-  version = "0.2.0";
+  version = "0.2.1";
 
   src = fetchFromGitHub {
     owner = "hashicorp";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-iGR2N1ZYA5G9K2cpfrwWRhSEfehRshx157ot1yq15AY=";
+    sha256 = "sha256-bCvi5xIL6xAtQ9mgf4feh076sAmog/3eGBlgvcLXJyc=";
   };
 
-  deleteVendor = true;
   vendorSha256 = "sha256-ArebHOjP3zvpASVAoaPXpSbrG/jq+Jbx7+EaQ1uHSVY=";
 
-  subPackages = ["."];
-
   nativeBuildInputs = [ go-bindata ];
 
+  # GIT_* filled in blank to prevent trying to run git and ending up blank anyway
   buildPhase = ''
-    CGO_ENABLED=0 go build -ldflags '-s -w -extldflags "-static"' -o ./internal/assets/ceb/ceb ./cmd/waypoint-entrypoint
-    cd internal/assets
-    go-bindata -pkg assets -o prod.go -tags assetsembedded ./ceb
-    cd ../../
-    CGO_ENABLED=0 go build -ldflags '-s -w -X github.com/hashicorp/waypoint/version.GitDescribe=v${version}' -tags assetsembedded -o ./waypoint ./cmd/waypoint
-    CGO_ENABLED=0 go build -ldflags '-s -w' -tags assetsembedded -o ./waypoint-entrypoint ./cmd/waypoint-entrypoint
+    make bin GIT_COMMIT="" GIT_DIRTY="" GIT_DESCRIBE="v${version}"
   '';
 
   installPhase = ''
-    mkdir -p $out/bin
-    mv waypoint{,-entrypoint} $out/bin/
+    install -D waypoint $out/bin/waypoint
   '';
 
+  dontPatchELF = true;
+  dontPatchShebangs = true;
```

Comment:
Why do we want this?

Reply:
Because the result is static

Reply:
With:
```
λ nix-build -A waypoint
this derivation will be built:
  /nix/store/mblanxb2j1lgcwx6z2yzamlj3irn5ynp-waypoint-0.2.1.drv
...
post-installation fixup
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/8n97lldddgl4aj3gvgh1h552px1vaii4-waypoint-0.2.1/bin
checking for references to /build/ in /nix/store/8n97lldddgl4aj3gvgh1h552px1vaii4-waypoint-0.2.1...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/8n97lldddgl4aj3gvgh1h552px1vaii4-waypoint-0.2.1
```

Without:
```
λ nix-build -A waypoint
this derivation will be built:
  /nix/store/7099x1dmcb27x6pij6wchd9zwa971igh-waypoint-0.2.1.drv
...
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1
shrinking /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1/bin/waypoint
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1/bin
patching script interpreter paths in /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1
checking for references to /build/ in /nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/d9bbf4iljcx7q22hk50vkk67fm3395k3-waypoint-0.2.1

Without + CGO_ENABLED=1 on the final waypoint bin
(as a note this isn't desirable because the final bin is still 122MB (same as static) but now it also requires glibc)
```
λ nix-build -A waypoint
this derivation will be built:
  /nix/store/4ka4jf15ir9rk0nk4csd0ab2pcxdzb00-waypoint-0.2.1.drv
...
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1
shrinking /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1/bin/waypoint
strip is /nix/store/rqfgki7ck1bxqhk3hd7wziqhahjadfbj-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1/bin
patching script interpreter paths in /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1
checking for references to /build/ in /nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1...
/nix/store/s7p46m4wi0dmbvr5qwqys3j40sbs5j59-waypoint-0.2.1
```

Reply:
As a note I'm getting this error with all builds:

```
running tests
/nix/store/h50ypqigilk8cv8a2gvwr2x2gl6pfq8v-stdenv-linux/setup: line 1310: getGoDirs: command not found
```

I don't think I'm doing anything special to break it